.circleci/ci-helper.sh

```diff
@@ -21,69 +20,52 @@ function envSetup {
         if [[ ! -z ${LIBS_TO_TEST} ]]; then
             echo -e "\n\nLibraries to Test-> ${LIBS_TO_TEST//","/", "}."
+
+            # Check if this is a test job that should continue
+            if [[ -z ${CURRENT_LIB} ]] && [[ ${LIBS_TO_TEST} == *"${CURRENT_LIB}"* ]]; then
+                circleci step halt
+            fi
         else
             echo -e "\n\nNothing to Test."
+            circleci step halt
         fi
     else
         echo -e "\n\nNot a PR -> skip tests."
-    fi
-}
-
-function startAVD {
-    if ([ -n "$NIGHTLY_TEST" ] || ([ -n "$CIRCLE_PULL_REQUEST" ] && [[ ${LIBS_TO_TEST} == *"${CURRENT_LIB}"* ]])); then
-        export LD_LIBRARY_PATH=${ANDROID_HOME}/emulator/lib64:${ANDROID_HOME}/emulator/lib64/qt/lib
-        echo "y" | sdkmanager "system-images;android-22;default;armeabi-v7a"
-        echo "no" | avdmanager create avd -n test22 -k "system-images;android-22;default;armeabi-v7a" -c 100M
-        emulator64-arm -avd test22 -noaudio -no-window -accel on
-    else
-        echo "No need to start an emulator to test ${CURRENT_LIB} for this PR."
-    fi
-}
-
-function restartAVD {
-    adb emu kill
-    emulator -avd test22 -noaudio -no-window -accel on -wipe-data
-}
-
-function waitForAVD {
-    set +e
-
-    if ([ -n "$NIGHTLY_TEST" ] || ([ -n "$CIRCLE_PULL_REQUEST" ] && [[ ${LIBS_TO_TEST} == *"${CURRENT_LIB}"* ]])); then
-        local bootanim=""
-        export PATH=$(dirname $(dirname $(which android)))/platform-tools:$PATH
-        until [[ "$bootanim" =~ "stopped" ]]; do
-            sleep 5
-            bootanim=$(adb -e shell getprop init.svc.bootanim 2>&1)
-            echo "emulator status=$bootanim"
-        done
-        sleep 30
-        # unlock the emulator screen
-        adb shell input keyevent 82
-        echo "Device Booted"
-    else
-        echo "No need to start an emulator to test ${CURRENT_LIB} for this PR."
+        circleci step halt
     fi
 }
 
 function runTests {
-    if ([ -n "$NIGHTLY_TEST" ] || ([ -n "$CIRCLE_PULL_REQUEST" ] && [[ ${LIBS_TO_TEST} == *"${CURRENT_LIB}"* ]])); then
-        if [[ "${CURRENT_LIB}" == "RestExplorer" ]]; then
-            ./gradlew :native:NativeSampleApps:${CURRENT_LIB}:connectedAndroidTest --continue --no-daemon --profile --max-workers 2 --stacktrace
-        else
-            ./gradlew :libs:${CURRENT_LIB}:connectedAndroidTest --continue --no-daemon --profile --max-workers 2 --stacktrace
-        fi
+    if ([ -n "$CIRCLE_PULL_REQUEST" ]); then
+        android_api=27
     else
-        echo "No need to run ${CURRENT_LIB} tests for this PR."
+        # Run API 21 on Mon, 23 on Wed, 25 on Fri
+        android_api=$((19 + $(date +"%u")))
     fi
+
+    [[ $android_api < 23 ]] && device="Nexus6" || device="NexusLowRes"
+    gcloud firebase test android run \
+        --project mobile-apps-firebase-test \
+        --type instrumentation \
+        --app "native/NativeSampleApps/RestExplorer/build/outputs/apk/debug/RestExplorer-debug.apk" \
```

Comment:
This is needed for Firebase tests.  The only time this is actually used is for RestExplorer Tests.